### PR TITLE
🪟 🐛 Fix validation when setting the connection stream cursor or primary key

### PR DESCRIPTION
**Pull request by edmundito**
## What
Related to https://github.com/airbytehq/oncall/issues/948

This fixes an issue related to validation not running when the connection stream primary key or cursor field is not set.

## How
The validation was happening in the wrong order because the form initial values streams were being sorted when the CatalogTree component rendered. Instead, now it makes a copy of the array before sorting since sorting is one of those weird [functions that is done in place](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort).

### CODE REVIEW

File: airbyte-webapp/src/pages/ConnectionPage/pages/ConnectionItemPage/ConnectionReplicationTab.tsx [hunk 1]
```diff
@@ -90,12 +90,8 @@ export const ConnectionReplicationTab: React.FC = () => {
       // This could be due to user changes (e.g. in the sync mode) or due to new/removed
       // streams due to a "refreshed source schema".
       const catalogHasChanged = !equal(
-        formValues.syncCatalog.streams
-          .filter((s) => s.config?.selected)
-          .sort(naturalComparatorBy((syncStream) => syncStream.stream?.name ?? "")),
-        connection.syncCatalog.streams
-          .filter((s) => s.config?.selected)
-          .sort(naturalComparatorBy((syncStream) => syncStream.stream?.name ?? ""))
```
Comment (tealjulia) [inline airbyte-webapp/src/pages/ConnectionPage/pages/ConnectionItemPage/ConnectionReplicationTab.tsx:98]: I'm not sure I understand why these no longer need the sort.  The only place I see us sorting now is the catalogTree.  Isn't there still a chance that the order is not preserved between the connection object and the form values?

**Comment by tealjulia** [comment]
Tested locally and the bug does seem to be gone.  Just one question to make sure we're covering all our bases!

File: airbyte-webapp/src/pages/ConnectionPage/pages/ConnectionItemPage/ConnectionReplicationTab.tsx (hunk 1 above)
Comment (edmundito) [reply airbyte-webapp/src/pages/ConnectionPage/pages/ConnectionItemPage/ConnectionReplicationTab.tsx:98]: The reason why this was sorted in the first place was that the formValues data was getting mutated by the sort call in the CatalogTree component. The formValues now contain same stream order as received in the connection object, and the order should match.

**Review by teallarson — APPROVED** [review]
tested locally and problem is resolved, code looks good.

**Pull request merged**